File: Lua/fizzbuzz.lua
#!/usr/local/bin/lua

local f = assert(io.open(arg[1], "r"))
while true do
  local n1, n2, n3 = f:read("*number", "*number", "*number")
  if not n1 then break end
  local result = ''
  local n = 0
  while n < n3 do
    n = n + 1
    if (n % n1 == 0) then result = result .. "F" end
    if (n % n2 == 0) then result = result .. "B" end
    if (n % n1 ~= 0 and n % n2 ~= 0) then result = result .. n end
    result = result .. ' '
  end
  print (string.sub(result, 1, -2))
end
f:close()

File: Perl/fizzbuzz.pl
#!/usr/bin/perl

use warnings;
use strict;

my $filename = $ARGV[0];
open(my $FH, '<', $filename) || die($!);
while (my $line = <$FH>) {
	my $n = 0;
	my $result = '';
	if ($line =~ /^(\d+) (\d+) (\d+)$/) {
		while ($n < $3){		
			$n += 1;
			if ($n % $1 == 0) {
				$result = $result . "F";
			}
			if ($n % $2 == 0) {
				$result = $result . "B";
			}
			if ($n % $2 != 0 && $n % $1 != 0) {
				$result = $result . $n;
			}
			$result = $result . ' ';
		}
	chop $result;
	print "$result\n";
	}
}
